lib/optparse.rb

```diff
@@ -2048,7 +2062,7 @@ def inspect
     # Default stringizing method to emit standard error message.
     #
     def message
-      reason + ': ' + args.join(' ')
+      reason + ': ' + args.join(" ").gsub(/\s+$/, "")
```

Comment:
``gsub(/\s+$/, "")`` seems to do the same as ``rstrip``.

Reply:
No, the two are not the same.  `gsub` is actually intended.

```
irb(main):001:0> "foo  \nbar  \n".rstrip
=> "foo  \nbar"
irb(main):002:0> "foo  \nbar  \n".gsub(/\s+$/, "")
=> "foo\nbar"
```

Reply:
Sorry, I forgot about multiline match.